opal/datatype/opal_datatype_memcpy.h

```diff
@@ -14,7 +14,30 @@
 #ifndef OPAL_DATATYPE_MEMCPY_H_HAS_BEEN_INCLUDED
 #define OPAL_DATATYPE_MEMCPY_H_HAS_BEEN_INCLUDED
 
-#define MEMCPY( DST, SRC, BLENGTH ) \
-    memcpy( (DST), (SRC), (BLENGTH) )
+/*
+ * This macro is called whenever we are packing/unpacking a DDT that
+ * that is built with basic datatypes.
+ * Specifying a fixed size for the memcpy() makes the intel compiler
+ * inline it as an assignment operation.
+ * This code is a bit hacky, but doing this we can divide the latency
+ * by up to 2 during DDT exechanges.
+ */
+#define MEMCPY( DST, SRC, BLENGTH )                                 \
+    do {                                                            \
+        if( (BLENGTH) < 16 ) {                                      \
+            uintptr_t align = ((uintptr_t)(DST)) ^ ((uintptr_t)(SRC));                        \
```

Comment:
Are you really sure there will be a gain assigning instead of memcpy'ing, if both the src and dst are not aligned, even if it is the same misalignment?
Wondering whether we shouldn't replace the "^" by a "|"